proof/refine/RISCV64/KHeap_R.thy

```diff
@@ -949,6 +949,11 @@ proof -
                   elim: cte_wp_atE' canonical_address_add)
 qed
 
+lemma obj_at'_is_canonical:
+  "\<lbrakk>pspace_canonical' s; obj_at' P t s\<rbrakk> \<Longrightarrow> canonical_address t"
```

Comment:
I'm fairly sure we don't have `canonical_address` for ARM. Having a look at the places where this rule is used, `simp` might be most useful, so I think I'll do that unless there are any objections

Reply:
Ah, looks like you're right and it's currently only for RISCV64 and X64. That probably explains why I'm not familiar with it.

Reply:
Yes, it's a 64bit concept. I would not recommend `[simp]` for this, because that fires too often. You will often have `canonical` in the goal, but no `obj_at` or `pspace_canonical`